proposals/large-ipld-dags.md

```diff
@@ -0,0 +1,116 @@
+   data for both storage and retrieval. This is especially true when interacting with IPFS.
+2. This workaround requires storing an "overlay" DAG in Filecoin (paying for that storage).
+
+Second, it should be possible to retrieve subsets of DAGs. While the underlying protocols support
+retrieving subsets of DAGs, the CLI does not. This makes it impossible to, e.g., retrieve a single
+file from a directory without modifying Lotus.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+
+There is no easy way (e.g., no out-of-band deals) to store large (> sector size) IPLD DAGs while
+preserving the DAG structure.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+
+* `lotus client deal` should accept an IPLD selector.
+* `lotus cleint deal` should automatically split large DAGs between multiple sectors.
+* `lotus client retrieve` should support retrieving IPLD selectors (dag subsets).
+
+#### Impact
+_How directly important is the outcome to web3 dev stack product-market fit?_
+
+🔥
+
+At the moment, any tool wishing to support storing IPFS files/directories larger than 32GiB will need to store these IPFS files/directories as "raw blocks", throwing away all the DAG structural information. This will make future retrieval deals for subsets of this data infeasible and will make IPFS interop extremely difficult.
+
+This is only one 🔥 because there are plenty of useful sub-32GiB datasets and non-IPFS datasets.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+🎯🎯🎯
+
+If we don't solve this now, users will likely store large DAGs any way they can (e.g., as raw
+blocks). We could end up with a lot of unfortunately structured data in Filecoin that's difficult to
+retrieve and work with, especially from IPFS.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+??
+
+## Project definition
+#### Brief plan of attack
+
+1. Implement selector support in `lotus client deal`. 
+2. Implement selector support in `lotus client retrieve`.
+3. Support automatically splitting large dags into across deals in `lotus client retrieve`. 
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+All three of the above commands have been implemented.
+
+NOTE: stopping anywhere along the way will yield a useful result. As long as the first step is finished (selector support for `lotus client deal`) the 
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+1. Developers can easily store large directory trees on Filecoin.
+2. Developers can easily retrieve individual files from large datasets on Filecoin.
+
```

Comment:
Archive like  [wikipedia_en_all_maxi_2021-02.zim](https://download.kiwix.org/zim/wikipedia/wikipedia_en_all_maxi_2021-02.zim)  is ~80GB, we unpack it and put unixfs DAG on IPFS together with original ZIM, so the final DAG  is 4 or 5 times the original size (ignoring deduplication).
Sounds like a good test case to use for validating deliverable here:

```suggestion
3. Snapshots of English Wikipedia can be stored on Filecoin.
```